packages/compiler/src/experimental/mutators.ts

```diff
@@ -0,0 +1,389 @@
+import { Program } from "../core/program.js";
+import { isArrayModelType } from "../core/type-utils.js";
+import {
+  Decorator,
+  DecoratorFunction,
+  Enum,
+  EnumMember,
+  FunctionParameter,
+  FunctionType,
+  Interface,
+  IntrinsicType,
+  Model,
+  ModelProperty,
+  Namespace,
+  ObjectType,
+  Operation,
+  Projection,
+  Scalar,
+  ScalarConstructor,
+  StringTemplate,
+  StringTemplateSpan,
+  SyntaxKind,
+  TemplateParameter,
+  Tuple,
+  Type,
+  Union,
+  UnionVariant,
+} from "../core/types.js";
+import { CustomKeyMap } from "../emitter-framework/custom-key-map.js";
+import { $doc, isVisible } from "../lib/decorators.js";
+import { Realm } from "./realm.js";
+import { createTypeFactory } from "./type-factory.js";
+
+export type MutatorRecord<T extends Type> =
```

Comment:
Could be worth adding `@experimental` to exported elements(at least the functions so it shows that way in the docs too)
https://typedoc.org/tags/experimental/